plugins/CoreAdminHome/javascripts/generalSettings.js

```diff
@@ -113,20 +116,31 @@ $(document).ready(function () {
 
     $("#logoUploadForm").submit(function (data) {
         var submittingForm = $(this);
+        var isSubmittingLogo = ($('#customLogo').val() != '')
+        var isSubmittingFavicon = ($('#customFavicon').val() != '')
         $('.uploaderror').fadeOut();
         var frameName = "upload" + (new Date()).getTime();
         var uploadFrame = $("<iframe name=\"" + frameName + "\" />");
         uploadFrame.css("display", "none");
         uploadFrame.load(function (data) {
             setTimeout(function () {
-                refreshCustomLogo();
-
                 var frameContent = $(uploadFrame.contents()).find('body').html();
                 frameContent = $.trim(frameContent);
 
                 if ('0' === frameContent) {
                     $('.uploaderror').show();
                 }
+                else {
+                    // Upload succeed, so we update the images availability
+                    // according to what have been uploaded
+                    if(isSubmittingLogo) {
```

Comment:
just a minor thing, do you mind adding a whitespace before the opening brackets?